File: Jfml_Esempio/src/jfml_esempio/CasiDiTest_IrisMamdani2.java
package jfml_esempio;

import java.io.File;

import jfml.FuzzyInferenceSystem;
import jfml.JFML;
import jfml.knowledgebase.variable.KnowledgeBaseVariable;

public class CasiDiTest_IrisMamdani2 {

	public static void main(String[] args) {
		// Loading Fuzzy System from an XML file according the standard IEEE
		// 1855
		File xml = new File("./XMLFiles/IrisMamdani2.xml");
		FuzzyInferenceSystem fs = JFML.load(xml);
		float[] limitiSL = { 4.3f, 7.9f, 6.1f };
		float[] limitiSW = { 2f, 4.4f, 3.2f };
		float[] limitiPL = { 1f, 6.9f, 3.9f };
		float[] limitiPW = { 0.1f, 2.5f, 1.2f };
		// Set inputs values SepalLength
		KnowledgeBaseVariable fuzzyVarSP;

		// Set inputs values SepalWidth
		KnowledgeBaseVariable fuzzyVarSW;

		// Set inputs values PetalLength
		KnowledgeBaseVariable fuzzyVarPL;

		// Set inputs values PetalWidth
		KnowledgeBaseVariable fuzzyVarPW;

		for (int i = 0; i < limitiSL.length; i++) {
			fuzzyVarSP = fs.getVariable("SepalLength");
			fuzzyVarSP.setValue(limitiSL[i]);
			for (int j = 0; j < limitiSW.length; j++) {
				fuzzyVarSW = fs.getVariable("SepalWidth");
				fuzzyVarSW.setValue(limitiSW[j]);
				for (int j2 = 0; j2 < limitiPL.length; j2++) {
					fuzzyVarPL = fs.getVariable("PetalLength");
					fuzzyVarPL.setValue(limitiPL[j2]);
					for (int k = 0; k < limitiPW.length; k++) {
						fuzzyVarPW = fs.getVariable("PetalWidth");
						fuzzyVarPW.setValue(limitiPW[k]);
						// Fuzzy inference test System Fuzzy
						fs.evaluate();

						// Get output test System Fuzzy
						KnowledgeBaseVariable irisClass = fs.getVariable("irisClass");
						float value = irisClass.getValue();

						// Printing results System Fuzzy
						System.out.println("RESULTS System Fuzzy");
						System.out.println(" (INPUT): " + fuzzyVarSP.getName() + "=" + fuzzyVarSP.getValue()
								+ fuzzyVarSW.getName() + "=" + fuzzyVarSW.getValue() + fuzzyVarPL.getName() + "="
								+ fuzzyVarPL.getValue() + fuzzyVarPW.getName() + "=" + fuzzyVarPW.getValue());
						System.out.println(" (OUTPUT): " + irisClass.getName() + "=" + value);

						// Printing the FuzzySystem System Fuzzy test
						System.out.println(fs.toString());
					}
				}
			}
		}
		/*
		 * // Fuzzy inference test System Fuzzy fs.evaluate();
		 * 
		 * // Get output test System Fuzzy KnowledgeBaseVariable irisClass =
		 * fs.getVariable("irisClass"); float value = irisClass.getValue();
		 * 
		 * // Printing results System Fuzzy System.out.println(
		 * "RESULTS System Fuzzy"); System.out.println(" (INPUT): " +
		 * fuzzyVarSP.getName() + "=" + fuzzyVarSP.getValue() +
		 * fuzzyVarSW.getName() + "=" + fuzzyVarSW.getValue() +
		 * fuzzyVarPL.getName() + "=" + fuzzyVarPL.getValue() +
		 * fuzzyVarPW.getName() + "=" + fuzzyVarPW.getValue());
		 * System.out.println(" (OUTPUT): " + irisClass.getName() + "=" +
		 * value);
		 * 
		 * // Printing the FuzzySystem System Fuzzy test
		 * System.out.println(fs.toString());
		 */
	}

}

File: Py4jfml/testlib/CasiDiTestIrisMamdani2.py
from py4j.java_gateway import JavaGateway
from py4jfml.Py4Jfml import Py4jfml
import os

gateway = JavaGateway()

#percorso del file xml da valutare
str_xml = os.getcwd()+"/"+ "IrisMamdani2.xml"


#Sistema fuzzy
fs = Py4jfml.load(str_xml)
#limite minimo massimo e medio SepalLength
limitiSL = [4.3, 7.9, 6.1]

#limite minimo massimo e medio SepalWidth
limitiSW = [2., 4.4, 3.2]

#limite minimo massimo e medio PetalLength
limitiPL = [1., 6.9, 3.9]

#limite minimo massimo e medio PetalWidth
limitiPW = [0.1, 2.5, 1.2]

for i in range(3):
    fuzzyVarSL = fs.getVariable("SepalLength")

    #imposto i valori
    fuzzyVarSL.setValue(float(limitiSL[i]))
    for j in range(3):
        fuzzyVarSW = fs.getVariable("SepalWidth")

        # imposto i valori
        fuzzyVarSW.setValue(float(limitiSW[j]))
        for j2 in range(3):
            fuzzyVarPL = fs.getVariable("PetalLength")

            # imposto i valori
            fuzzyVarPL.setValue(float(limitiPL[j2]))
            for k in range(3):
                fuzzyVarPW = fs.getVariable("PetalWidth")

                # imposto i valori
                fuzzyVarPW.setValue(float(limitiPW[k]))

                #valuto il sistema fuzzy
                fs.evaluate()

                #ottengo valori di output
                irisClass = fs.getVariable("irisClass")
                value = irisClass.getValue()

                # stampo i risultati del sistema fuzzy
                print("RESULTS System Fuzzy")
                print(" (INPUT): " + str(fuzzyVarSL.getName()) + "=" + str(fuzzyVarSL.getValue()) + str(fuzzyVarSW.getName()) + "=" + str(fuzzyVarSW.getValue()) + str(fuzzyVarPL.getName()) + "=" + str(fuzzyVarPL.getValue()) + str(fuzzyVarPW.getName()) + "=" + str(fuzzyVarPW.getValue()))
                print(" (OUTPUT): " + str(irisClass.getName()) + "=" + str(value))

                # stampo il sistema fuzzy
                print("System Fuzzy: ")
                print(fs)
